### DISCUSSION TITLE
Random AWS nodes reboot

### DISCUSSION THREAD

**Discussion by yaroslavkasatikov** [okd-project/okd#1303] [General]
Hello team,
We faced to strange issue after upgrade 4.7 -> 4.8 and 4.8 -> 4.9

When we upgraded to 4.8.0-0.okd-2021-10-24-061736, we found that some nodes (Our cluster located in AWS) randomly rebooted. We found that it is related to t3 and t3a nodes (doesn't matter 2xlarge or xlarge). We have something like this:

```
```Events:
  Type     Reason                   Age                From     Message
  ----     ------                   ----               ----     -------
  Normal   Starting                 42m                kubelet  Starting kubelet.
  Normal   NodeHasSufficientMemory  42m (x2 over 42m)  kubelet  Node ip-10-0-181-231.eu-central-1.compute.internal status is now: NodeHasSufficientMemory
  Normal   NodeHasNoDiskPressure    42m                kubelet  Node ip-10-0-181-231.eu-central-1.compute.internal status is now: NodeHasNoDiskPressure
  Normal   NodeHasSufficientPID     42m                kubelet  Node ip-10-0-181-231.eu-central-1.compute.internal status is now: NodeHasSufficientPID
  Warning  Rebooted                 42m                kubelet  Node ip-10-0-181-231.eu-central-1.compute.internal has been rebooted, boot id: c56f146b-4d9f-4e8c-b8f1-89a7897d8545
  Normal   Starting                 41m                kubelet  Starting kubelet.
  Normal   NodeAllocatableEnforced  41m                kubelet  Updated Node Allocatable limit across pods
  Normal   NodeHasSufficientMemory  41m                kubelet  Node ip-10-0-181-231.eu-central-1.compute.internal status is now: NodeHasSufficientMemory
  Normal   NodeHasNoDiskPressure    41m                kubelet  Node ip-10-0-181-231.eu-central-1.compute.internal status is now: NodeHasNoDiskPressure
  Normal   NodeHasSufficientPID     41m                kubelet  Node ip-10-0-181-231.eu-central-1.compute.internal status is now: NodeHasSufficientPID
  Normal   NodeReady                41m                kubelet  Node ip-10-0-181-231.eu-central-1.compute.internal status is now: NodeReady```

```
Sometimes nodes rebooted and became ready, but sometimes they stuck in NotReady and only termination on AWS helped.

As result, we upgraded cluster to  4.9.0-0.okd-2022-01-14-230113 but the behavior didn't changed. 

Could you please help me with the issue.. Please write which additional info can help in investigation, unfortunately, I don't know which logs can be useful.. 
I will be happy to attach it.

```
```[tf@yaroslav-new tlb.payment]$ oc get clusterversion
NAME      VERSION                         AVAILABLE   PROGRESSING   SINCE   STATUS
version   4.9.0-0.okd-2022-01-14-230113   True        False         3h24m   Cluster version is 4.9.0-0.okd-2022-01-14-230113```

Kubernetes: v1.22.3+e790d7f
```Fedora release 34 (Thirty Four)
NAME=Fedora
VERSION="34 (CoreOS)"
ID=fedora
VERSION_ID=34
VERSION_CODENAME=""
PLATFORM_ID="platform:f34"
PRETTY_NAME="Fedora CoreOS 34"
ANSI_COLOR="0;38;2;60;110;180"
LOGO=fedora-logo-icon
CPE_NAME="cpe:/o:fedoraproject:fedora:34"
HOME_URL="https://getfedora.org/coreos/"
DOCUMENTATION_URL="https://docs.fedoraproject.org/en-US/fedora-coreos/"
SUPPORT_URL="https://github.com/coreos/fedora-coreos-tracker/"
BUG_REPORT_URL="https://github.com/coreos/fedora-coreos-tracker/"
REDHAT_BUGZILLA_PRODUCT="Fedora"
REDHAT_BUGZILLA_PRODUCT_VERSION=34
REDHAT_SUPPORT_PRODUCT="Fedora"
REDHAT_SUPPORT_PRODUCT_VERSION=34
PRIVACY_POLICY_URL="https://fedoraproject.org/wiki/Legal:PrivacyPolicy"
VARIANT="CoreOS"
VARIANT_ID=coreos
OSTREE_VERSION='49.34.202201141610-0'
DEFAULT_HOSTNAME=localhost
Fedora release 34 (Thirty Four)
Fedora release 34 (Thirty Four)
cpe:/o:fedoraproject:fedora:34```

```
Thanks,
Yaroslav

**Comment by vrutkovs** [comment]
Please attach (or upload to the public file sharing service) [must-gather](https://docs.okd.io/latest/support/gathering-cluster-data.html#support_gathering_data_gathering-cluster-data) archive

Also, is it reproducible on OKD 4.11?

**Reply by yaroslavkasatikov** [reply]
Hi Vadim,
Unfortunately, I haven't updated cluster up to 4.11 yet. As I understood correct, I can update it only up to 4.9.0-0.okd-2022-02-12-140851 and upgrade step-by-step. The cluster upgrade is rather painful exercise for our cluster, because databases nodes reboot. If you know a way how to directly upgrade cluster to 4.11 - please tell me.

I'm started must-gather on 4.9 cluster. I have the second cluster 4.8.0-0.okd-2021-10-24-061736 with the same symptoms. Only t3* nodes affected too.

**Reply by vrutkovs** [reply]
>If you know a way how to directly upgrade cluster to 4.11 - please tell me.

No such way, you'd have to upgrade to 4.9 then to 4.10 and then to 4.11, sorry.

**Comment by yaroslavkasatikov** [comment]
@vrutkovs here is must-gather tar https://dropmefiles.com/J42mq for 4.9

**Reply by vrutkovs** [reply]
Not seeing anything which would reboot these nodes on purpose, so most likely its a Fedora kernel bug. 4.9 uses no-longer-supported F34 so you may want to update to 4.11 and report the bug for F36 if it still happens

**Comment by yaroslavkasatikov** [comment]
https://dropmefiles.com/DWmmH for 4.8

**Comment by msteenhu** [comment]
We have the same issue on our  4.9.0-0.okd-2022-02-12-140851 production cluster on bare metal machines, latest Intel generation of CPU and MB.
Once I could dig up kernel panic log in the journal that points to a vague kernel bug. But usually journal just shows the reboot.

Usually the host reboots but we also had hosts that just lock up, so k8s cannot recover "ReadWriteOnce" VPCs until we manually reboot (or install some operator that detects node lockup).

We are planning upgrades hoping to solve this kernel instability. I would say we have every 2 months or so a host crash.

**Comment by yaroslavkasatikov** [comment]
Hi @vrutkovs , hi team,
I have upgraded one cluster upto the latest version (4.11) and reboots have gone (there is no reboot for 24h). Will wait several days and report. 

So yes, seems like OS bug which fixed with upgrade.
Thanks!

**Comment by yaroslavkasatikov** [comment]
Cluster upgrading helped with reboots, but lead to new issue - https://github.com/openshift/okd/issues/1310
